golem-cli/src/oss/main.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::config::OssProfile;
+use crate::examples;
+use crate::factory::ServiceFactory;
+use crate::oss::command::{GolemOssCommand, OssCommand};
+use crate::oss::factory::OssServiceFactory;
+use crate::oss::model::OssContext;
+use crate::stubgen::handle_stubgen;
+use colored::Colorize;
+use std::path::PathBuf;
+
+pub async fn async_main(
+    cmd: GolemOssCommand,
+    profile: OssProfile,
+    config_dir: PathBuf,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let OssProfile {
+        url,
+        worker_url,
+        allow_insecure,
+        config,
+    } = profile;
+
+    let format = cmd.format.unwrap_or(config.default_format);
+
+    let component_url = url;
+    let worker_url = worker_url.unwrap_or_else(|| component_url.clone());
```

Comment:
is there reason why property for component url is called once `url` and once `component_url`, and for worker just `worker_url` ?